tests/determine-reboot-cause_test.py

```diff
@@ -199,3 +204,34 @@ def test_determine_reboot_cause_main_with_reboot_cause_dir(self):
             determine_reboot_cause.main()
             assert os.path.exists("host/reboot-cause/reboot-cause.txt") == True
             assert os.path.exists("host/reboot-cause/previous-reboot-cause.json") == True   
+
+    def create_mock_platform_json(self, dpus):
+        """Helper function to create a mock platform.json file."""
+        os.makedirs(os.path.dirname(PLATFORM_JSON_PATH), exist_ok=True)
+        with open(PLATFORM_JSON_PATH, "w") as f:
+            json.dump({"DPUS": dpus}, f)
+
+    @mock.patch('sonic_py_common.device_info.is_smartswitch', return_value=True)
+    @mock.patch('sonic_py_common.device_info.get_platform', return_value='some_platform')
+    def test_check_and_create_dpu_dirs(self, mock_get_platform, mock_is_smartswitch):
+        # Call the function under test
+        result = check_and_create_dpu_dirs()
+
+    @mock.patch('sonic_py_common.device_info.get_platform_info', return_value={'platform': 'some_platform'})
+    @mock.patch('sonic_py_common.device_info.is_smartswitch', return_value=True)
+    @mock.patch('os.path.exists')
+    @mock.patch('builtins.open', new_callable=mock.mock_open, read_data='{"DPUS": ["dpu0", "dpu1"]}')
+    @mock.patch('os.makedirs')
+    def test_check_and_create_dpu_dirs_with_platform_json(self, mock_makedirs, mock_open, mock_exists, mock_is_smartswitch, mock_get_platform_info):
+        # Mock the platform.json existence
+        mock_exists.side_effect = lambda path: path == "/usr/share/sonic/device/some_platform/platform.json"
+
+        # Call the function under test
+        check_and_create_dpu_dirs()
+
+        # Assert that open was called correctly
+        mock_open.assert_called_once_with("/usr/share/sonic/device/some_platform/platform.json", 'r')
+
+        # Assert that makedirs was called for the DPU directories
+        mock_makedirs.assert_any_call(os.path.join('/host/reboot-cause/module', 'dpu0'))
+        mock_makedirs.assert_any_call(os.path.join('/host/reboot-cause/module', 'dpu1'))
```

Comment:
`read_dpu_reboot_cause_files_and_save_chassis_state_d` is not covered by the tests. Please new test for it.

Reply:
@oleksandrivantsiv   will do but can we defer it as the entire files doe not have a test yet.